utils/aplay/aplay.c

```diff
@@ -555,7 +611,11 @@ static void *io_worker_routine(struct io_worker *w) {
 			if (ret % pcm_format_size != 0)
 				warn("Invalid read from BlueALSA source PCM: %zd %% %zd != 0", ret, pcm_format_size);
 
-			ffb_seek(&buffer, read_samples);
+#if WITH_LIBSAMPLERATE
+			if (use_resampler)
+				resampler_convert_to_native_endian_format(&read_buffer.tail, read_samples, pcm_format);
```

Comment:
```suggestion
				resampler_convert_to_native_endian_format(read_buffer.tail, read_samples, pcm_format);
```

This was causing segfault :D

Reply:
Anyway there is still a bug with aptx-HD because the `/* convert to S32 */` is wrong. We need to shift `data[n] <<= 8;` otherwise audio is VEEEERRYY quiet. I will push these changes later.